src/repo.rs

```diff
@@ -802,6 +802,33 @@ impl Repository {
         }
     }
 
+    /// Create a stream to write blob
+    ///
+    /// This function may need to buffer the data on disk and will in general
+    /// not be the right choice if you know the size of the data to write.
+    ///
+    /// Use `BlobWriter::commit()` to commit the write to the object db
+    /// and get the object id.
+    ///
+    /// If the `hintpath` parameter is filled, it will be used to determine
+    /// what git filters should be applied to the object before it is written
+    /// to the object database.
+    pub fn blob_writer(&self, hintpath: Option<&Path>) -> Result<BlobWriter, Error> {
+        let path_str = match hintpath {
+            Some(path) => Some(try!(path.into_c_string())),
+            None => None,
+        };
+        let path = match path_str {
+            Some(path) => path.as_ptr(),
```

Comment:
This'll actually want to be `ref path` to avoid the use-after free I believe

Reply:
Do you mean `Some(ref path)` here?

Reply:
Indeed!